ironmon_tracker/Program.lua

```diff
@@ -31,6 +31,97 @@ Program.GameData = {
 	},
 }
 
+Program.GameTimer = {
+	timeLastChecked = 0, -- Number of seconds
+	showPauseTipUntil = 0, -- Displays "how to pause timer" tip until this time occurs; number of seconds
+	hasStarted = false, -- Used to determine if the game has started (not in Title menus)
+	isPaused = false, -- Used to manually pause the timer
+	readyToDraw = false, -- Anytime the timer changes, it needs to be redrawn
+	textColor = 0xFFFFFFFF,
+	pauseColor = 0xFFFFFF00,
+	boxColor = 0x78000000,
+	margin = 0,
+	padding = 2,
+	box = {
+		x = Constants.SCREEN.WIDTH,
+		y = Constants.SCREEN.HEIGHT,
+		width = 20,
+		height = Constants.Font.SIZE,
+	},
+	getText = function(self)
+		return Utils.formatTime(Tracker.Data.playtime or 0)
+	end,
+	initialize = function(self)
+		self.hasStarted = false
+		self.box.height = Constants.Font.SIZE - 4 + (2 * self.padding)
+		self.box.y = math.max(Constants.SCREEN.HEIGHT - self.box.height - self.margin - 1, 0)
+		self:update()
+	end,
+	start = function(self)
+		self.hasStarted = true
+		self.isPaused = false
+		self.timeLastChecked = os.time()
+		self.showPauseTipUntil = self.timeLastChecked
+	end,
+	pause = function(self)
+		if not self.hasStarted then return end
+		self.isPaused = true
+	end,
+	unpause = function(self)
+		if self.isPaused then
+			self.timeLastChecked = os.time()
+		end
+		self.isPaused = false
+	end,
+	update = function(self)
+		local currTime = os.time()
+		local prevTime = self.timeLastChecked
+		self.timeLastChecked = currTime
+		if self.hasStarted and not self.isPaused then
+			local timeDelta = math.floor(os.difftime(currTime, prevTime))
+			Tracker.Data.playtime = Tracker.Data.playtime + timeDelta
+			self.readyToDraw = (timeDelta ~= 0)
+		end
+		-- Update box, align bottom-right
+		self.box.width = Utils.calcWordPixelLength(self:getText() or "") - 1 + (2 * self.padding)
+		self.box.x = math.max(Constants.SCREEN.WIDTH - self.box.width - self.margin - 1, 0)
+	end,
+	reset = function(self)
+		Tracker.Data.playtime = 0
+		self.hasStarted = false
+		self.readyToDraw = false
+	end,
+	checkInput = function(self, xmouse, ymouse)
+		local clicked = Input.isMouseInArea(xmouse, ymouse, self.box.x, self.box.y, self.box.width, self.box.height)
+		if clicked then
+			if self.isPaused then
+				self:unpause()
+			else
+				self:pause()
+			end
+			self.readyToDraw = true
+		end
+	end,
```

Comment:
This part doesn't check for if the play time is being displayed, meaning if someone happened to click down where the timer would be (can potentially happen by accident if they have another window over bizhawk for example) it will be paused in the background and if they decide to show it later in the run it won't have actually been updating

Reply:
Fixed with https://github.com/besteon/Ironmon-Tracker/pull/391/commits/6c6c76e237f83359055965313292a694cd6cb2e5